CHANGES.md

```diff
@@ -1,5 +1,122 @@
+indicating that the proxy should treat the traffic as opaque TCP instead of
+attempting protocol detection.  This allows the proxy to provide TCP metrics
+and mTLS for server-speaks-first protocols.  It also enables support for
+TCP traffic in the Multicluster extension.
+
+**Upgrade notes**: Please see the [upgrade
+instructions](https://linkerd.io/2/tasks/upgrade/#upgrade-notice-stable-2100).
+
+* Proxy
+  * Updated the proxy to use TLS version 1.3; support for TLS 1.2 remains
+    enabled for compatibility with prior proxy versions
+  * Improved support for server-speaks-first protocols by allowing ports to be
+    marked as opaque, causing the proxy to skip protocol detection.  Ports can
+    be marked as opaque by setting the `config.linkerd.io/opaque-ports`
+    annotation on the Pod and Service or by using the `--opaque-ports` flag with
+    `linkerd inject`
+  * Ports `25,443,587,3306,5432,11211` have been removed from the default skip
+    ports; all traffic through those ports is now proxied and handled opaquely
+    by default
+  * Fixed an issue that could cause the inbound proxy to fail meshed HTTP/1
+    requests from older proxies (from the stable-2.8.x vintage)
+  * Added a new `/shutdown` admin endpoint that may only be accessed over the
+    loopback network allowing batch jobs to gracefully terminate the proxy on
+    completion
+
+* Control Plane
+  * Removed all components and functionality related to visibility, tracing,
+    or multicluster.  These have been moved into extensions
+  * Changed the identity controller to receive the trust anchor via environment
+    variable instead of by flag; this allows the certificate to be loaded from a
+    config map or secret (thanks @mgoltzsche!)
+  * Added PodDisruptionBudgets to the control plane components so that they
+    cannot be all terminated at the same time during disruptions
+    (thanks @tustvold!)
+
+* CLI
+  * Changed the `check` command to include each installed extension's `check`
+    output; this allows users to check for proper configuration and installation
+    of Linkerd without running a command for each extension
+  * Moved the `metrics`, `endpoints`, and `install-sp` commands into subcommands
+    under the `diagnostics` command
+  * Added an `--opaque-ports` flag to `linkerd inject` to easily mark ports
+    as opaque.
+  * Added the `repair` command which will repopulate resources needed for
+    properly upgrading a Linkerd installation
+  * Added Helm-style `set`, `set-string`, `values`, `set-files` customization
+    flags for the `linkerd install` and `linkerd upgrade` commands
+  * Introduced the `linkerd identity` command, used to fetch the TLS certificates
+    for injected pods (thanks @jimil749)
+  * Removed the `get` and `logs` command from the CLI
+
+* Helm
+  * Changed many Helm values, please see the upgrade notes
+
+* Viz
+  * Updated the Web UI to only display the "Gateway" sidebar link when the
```

Comment:
Probably worth calling out that the CLI now has `linkerd viz` set of subcommands, including `linkerd viz install`.